python/lsst/afw/image/exposureSummary.py

```diff
@@ -0,0 +1,88 @@
+# This file is part of afw.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from dataclasses import dataclass, asdict, field
+import yaml
+
+from ..typehandling import Storable, StorableHelperFactory
+
+
+__all__ = ("ExposureSummary", )
+
+
+def _default_corners():
+    return [float('nan')]*4
+
+
+@dataclass
+class ExposureSummary(Storable):
+    _factory = StorableHelperFactory(__name__, "ExposureSummary")
```

Comment:
Since `StorableHelperFactory` has no documentation, I'm not sure I can say anything about its use. I gather that it's a generic `PersistableFactory` for anything that implements `_read`?

Reply:
This is what @jmeyers314 put together, I believe.  He helped me with https://github.com/lsst/afw/blob/master/tests/test_Storable.py#L156-L216 which is what this is based on.

Reply:
There's a comment here https://github.com/lsst/afw/blob/master/include/lsst/afw/typehandling/python.h#L115 describing it.  Sorry this didn't end up in doxygen.  My doxygen-fu is weak.  Is it possible to have the comment (which describes both `StorableHelperFactory::read` and `StorableHelper::write` together) be captured as is?  Or would one need to break it apart into read/write components to be captured?

Reply:
It is technically possible, but most such comments in the existing Stack code don't actually make sense outside the context of a particular code file and its layout (e.g., they refer to method parameters that don't exist, and you'd need to do a search to figure out where they do exist). This is especially true in this case because `StorableHelperPersistenceHelper` (which is what I assumed that comment was for) is an implementation detail and should be hidden, while both `StorableHelper` and `StorableHelperFactory` need to be visible to class authors.

I would recommend having separate documentation for both `StorableHelper` and `StorableHelperFactory` (_not_ just their `write` and `read` methods, because things like the `_factory` line above are non-obvious), and take advantage of the fact that code documentation allows linking. I'd be happy to write something myself if you're willing to review it for accuracy.

Reply:
Presumably, this should go on another ticket?